cypress/integration/proposal-booking.spec.ts

```diff
@@ -119,12 +119,51 @@ context('Proposal booking tests ', () => {
           .should('exist');
       });
 
+      it('should be able to add new time slot by drag and drop to calendar', () => {
+        cy.contains('Proposal ID');
+        cy.get('[data-cy="btn-close-dialog"]').click();
+
+        cy.finishedLoading();
+
+        cy.get(
+          '#instrument-calls-tree-view [role=treeitem] [role=group] [role=treeitem]'
+        )
+          .first()
+          .trigger('dragstart');
+
+        const slot = new Date(getHourDateTimeAfter(-25, 'hours')).toISOString();
+        cy.get(`.rbc-day-slot [data-cy='event-slot-${slot}']`).scrollIntoView();
+
+        // NOTE: It needs to be forced because all the event slots are covered by this element .rbc-events-container and it is not possible to execute drop on event-slot
+        cy.get(`.rbc-day-slot [data-cy='event-slot-${slot}']`).trigger('drop', {
+          force: true,
+        });
+
+        cy.finishedLoading();
+
+        cy.get('[data-cy="time-slot-booking-dialog"]').contains(
+          /Time slot booking/i
+        );
+
+        cy.get('[data-cy="time-slot-booking-dialog"]')
+          .get('[data-cy="btn-save"]')
+          .should('exist');
+
+        cy.get(
+          '[data-cy="time-slot-booking-dialog"] [data-cy="startsAtInfo"]'
+        ).should('include.text', getHourDateTimeAfter(-25, 'hours'));
+        cy.get(
+          '[data-cy="time-slot-booking-dialog"] [data-cy="endsAtInfo"]'
+        ).should('include.text', getHourDateTimeAfter(-24, 'hours'));
+      });
+
       it('Draft events should have opacity', () => {
         cy.get('[data-cy="btn-close-dialog"]').click();
 
         // NOTE: Using fixed proposal name and shortcode because they are inserted inside the database using a seeder and always will be the same for the test cases.
-        // .parent().parent() thing is because the react big calendar component always uses that structure around an event shown on the calendar.
+        // .parent().parent().parent() thing is because the react big calendar component always uses that structure around an event shown on the calendar.
```

Comment:
Have you tried looking into possibility to use `.closest(<selector>)`. Closest gets the first DOM element that matches the selector (whether it be itself or one of its ancestors).

Reply:
I can take a look if that will do the trick. Thank you.